ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/result/impl/PreventiveAndCurativesRaoResultImpl.java

```diff
@@ -281,7 +281,7 @@ public double getRelativeMargin(Instant optimizedInstant, FlowCnec flowCnec, Uni
     public double getFlow(Instant optimizedInstant, FlowCnec flowCnec, Side side, Unit unit) {
         FlowResult flowResult = getFlowResult(optimizedInstant, flowCnec);
         if (Objects.nonNull(flowResult)) {
-            return flowResult.getFlow(flowCnec, side, unit);
+            return flowResult.getFlow(flowCnec, side, unit, optimizedInstant);
```

Comment:
I think the method getStateOptimizedBefore also needs rewriting for curative instants (looking at the current code, it looks like it should be broken for anything but the first curative instant)